crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -1037,6 +1096,8 @@ unpackMemValue sym tpr (LLVMValZero tp) = unpackZero sym tp tpr
 unpackMemValue _sym (LLVMPointerRepr w) (LLVMValInt blk bv)
   | Just Refl <- testEquality (bvWidth bv) w
   = return $ LLVMPointer blk bv
+unpackMemValue _sym (BVRepr w) (LLVMValInt _ bv)
+  | Just Refl <- testEquality w (bvWidth bv) = return bv
```

Comment:
I just noticed this, and I'm skeptical.  What's the motivation here? Unpacking a load directly toward a `BVType` is something we I don't think we should be doing; and if we do decide to do that for some reason, we should be asserting that the block number is 0.

Reply:
Ok, I did not understand this use-case correctly. As we just talked about offline, we should never be calling `unpackMemValue` (through e.g. `doLoad`) and so this case should instead be throwing a panic. Instead, I will add some documentation on top of `doLoad` to explain the correct usage.